CommunityToolkit.HighPerformance/Properties/Polyfills/DoesNotReturnAttribute.cs

```diff
@@ -0,0 +1,13 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if !NETSTANDARD2_1_OR_GREATER
+
+namespace System.Diagnostics.CodeAnalysis;
+
+/// <summary>Applied to a method that will never return under any circumstance.</summary>
+[AttributeUsage(AttributeTargets.Method, Inherited = false)]
+public sealed class DoesNotReturnAttribute : Attribute { }
```

Comment:
This cannot be public, at best it should be internal. But regardless, I don't think we should add this in this PR, given it's not really needed and all the other throw helpers in the paclage are not using this attribute either. If we really wanted to do this, it should be a separate follow up PR just doing that 🙂

Reply:
Good point, I've removed the attribute.